### ISSUE TITLE
SMAPI errors when started via Vortex

### ISSUE THREAD

**Issue by Pickysaurus** [Nexus-Mods/Vortex#13230]
**Describe the bug**
From Vortex 1.6.11+ when the SMAPI is started via the Vortex shortcut on the dashboard with Stardew Valley Expanded installed there are numerous log errors that do not appear when launching SMAPI directly from the EXE. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Install these mods: https://next.nexusmods.com/stardewvalley/collections/d4vnoa/revisions/1
2. Launch SMAPI via Vortex. Start a new game. When the game attempts to save it will crash. 
3. Check your SMAPI log (Vortex Mods table) and see a log like this: https://smapi.io/log/7bd7baf7088345558772869816c4e59f?Levels=warn%7Eerror%7Ealert%7Ecritical
4. Start the game via SMAPI directly. Start a new game. When the game creates a save it will **not** crash. 
5. Verify the errors shown in the Vortex SMAPI log do not appear. 

**Expected behaviour**
Starting the game via Vortex and directly from the EXE should exhibit the same behaviour. 

**Screenshots**

**Platform (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Vortex Version [e.g. 0.16.12]

**Additional context**
Could this be related to the Electron version update in 1.6.11?

**Comment by Pickysaurus** [comment]
@pathoschild - Pinging you as these seems that it could be a weird behaviour of SMAPI under certain launch conditions.

**Comment by Pickysaurus** [comment]
DecidedlyHuman on the SDV Discord is working on a tool to dump the mod data loaded by SMAPI so we can compare what's different between the two launch methods. 
https://discord.com/channels/137344473976799233/156109690059751424/1026424350129274880

**Comment by Pickysaurus** [comment]
We've tested the files loaded by SMAPI in both cases and they are identical. This appears to be some kind of weirdness related to either how Vortex is starting SMAPI or how SMAPI/Content Patcher is behaving when launched via Vortex.

**Comment by TanninOne** [comment]
@Pickysaurus Could you point out which logs specifically only appear when run from Vortex?

**Comment by Pickysaurus** [comment]
> @Pickysaurus Could you point out which errors in the log specifically only appear when run from Vortex?

The URL filters to only show that already :)

**Comment by Pickysaurus** [comment]
atravita on the SDV Discord has been digging into this and we've not made much progress. 
https://discord.com/channels/137344473976799233/156109690059751424/1026466792761212968

**Comment by atravita-mods** [comment]
So, something is causing the tilesheets to load out of order when launched through Vortex.
<html><body>


12:20:16 | TRACE | Content Patcher | Patch "Stardew Valley Expanded > Load Maps/HaleyHouse" loaded Maps/HaleyHouse.
-- | -- | -- | --
12:20:16 | TRACE | SMAPI | Fixing tilesheet paths for map 'Assets/maps/locations/HaleyHouse.tbin' from mod 'Stardew Valley Expanded'...
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'townInterior' to 'Maps/townInterior'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'townInterior_2' to 'Maps/townInterior_2'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'VanillaFurniture.png' to 'Maps/VanillaFurniture'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'VanillaWallsAndFloors.png' to 'Maps/VanillaWallsAndFloors'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'paths' to 'Maps/paths'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'spring_z_extras.png' to 'Maps/spring_z_extras'.
12:20:16 | TRACE | SMAPI | Mapped tilesheet 'z_SVEbuildingShadow.png' to 'Maps/z_SVEbuildingShadow'.
12:20:16 | TRACE | SMAPI | Content Patcher loaded asset 'Maps/HaleyHouse' (for the 'Stardew Valley Expanded' content pack).
12:20:16 | ALERT | Content Patcher | mod tilesheets are: 1, 2, VanillaFurniture, VanillaWallsAndFloors, p, spring_z_extras, z_SVEbuildingShadow
12:20:16 | ALERT | Content Patcher | Mismatched IDs: VanillaFurniture != p
12:20:16 | WARN | Content Patcher | SMAPI found an issue with a 'Maps/HaleyHouse' map load: the 'Stardew Valley Expanded' content pack reordered the original tilesheets, which often causes crashes. Technical details for mod author: Expected order: 1, 2, p. See https://stardewvalleywiki.com/Modding:Maps#Tilesheet_order for help.


</body>
</html>

[Log](https://smapi.io/log/d34cd31ebc794aa3ab22473b6b8a9296?Levels=trace%7Edebug%7Einfo%7Ewarn%7Eerror%7Ealert%7Ecritical&Page=32&PerPage=1000) - scroll down to Maps/HaleyHouse.

Exactly why? No clue.

**Comment by Pickysaurus** [comment]
This issue related to another tool that appears to have stopped working in 1.6.11+ - so it could be related. https://github.com/Nexus-Mods/Vortex/issues/13200

**Comment by Pathoschild** [comment]
Tilesheets are loaded in the same order, but the text sort order changes to case-sensitive when run through Vortex:


&nbsp;         | without Vortex | with Vortex
-------------- | -------------- | -----------
thread culture | _invariant_    | _invariant_
UI culture     | `en-US` (or OS culture) | _invariant_
sample sort    | `example a-1, example A-2, EXAMPLE B` | `EXAMPLE B, example a-2, example A-1`

xTile automatically sorts tilesheets by ID, so this changes the map tilesheet order too. That in turn breaks the game's assumption that tilesheets are sorted case-insensitively when it references them by index.

This might be caused by https://github.com/Nexus-Mods/Vortex/commit/92afd2c3fd7dfa703a98dafcd9c809a29aa6a89f in Vortex 1.6.11, which sets the `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` environment variable.

**Comment by TanninOne** [comment]
Agreed, setting DOTNET_SYSTEM_GLOBALIZATION_INVARIANT globally was a bad idea and needs to be reverted anyway.

**Comment by Pickysaurus** [comment]
Thanks everyone!

https://github.com/Nexus-Mods/Vortex/commit/0ae919bd54cc48a16ad9219ec365aa453606079b

**Comment by TanninOne** [comment]
This should be fixed in 1.6.14
